File: Cap 12/12.5 - Encapsulamento/Efetivo/encapsu.kt
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.5
 * 
 * Programa ..: ENCAPSULAMENTO
 */

import java.util.*

class CLS_ALUNO {
  
    public var NOME: String = ""
    private var NOTAS: FloatArray = FloatArray(4)
    private var MEDIA: Float = 0f

    fun CMEDIA(): Float {
        var SOMA = 0f
        for (I in 0 until 4) {
            SOMA = SOMA + NOTAS[I]
        }
        MEDIA = SOMA / 4
        return MEDIA
    }

    fun PEGANOTA(POS: Int): Float {
        return NOTAS[POS]
    }

    fun PEGAMEDIA(): Float {
        return CMEDIA()
    }

    fun POENOTA(NT: Float, POS: Int) {
        NOTAS[POS] = NT
    }
    
}

fun main() {
  
    var entrada = Scanner(System.`in`)
    var ALUNO = CLS_ALUNO()
    var ENTRANOTA: Float

    println("DADOS ESCOLARES")
    println()

    print("Informe o nome: ")
    ALUNO.NOME = entrada.nextLine()

    println("Informe as notas: ")
    for (i in 0 until 4) {
        print("${i + 1}a. nota: ")
        ENTRANOTA = entrada.nextFloat()
        ALUNO.POENOTA(ENTRANOTA, i)
    }

    println()
    println("RELATORIO ESCOLAR")
    println()
    println("Nome: ${ALUNO.NOME}")
    for (i in 0 until 4) {
        println("${i + 1}a. nota: ${String.format("%5.1f", ALUNO.PEGANOTA(i))}")
    }
    println("Media: ${String.format("%5.1f", ALUNO.PEGAMEDIA())}")
	
}

File: Cap 12/12.5 - Encapsulamento/Efetivo/encapsu.ts
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.5
 * 
 * Programa ..: ENCAPSULAMENTO
 * 
 */

class CLS_ALUNO {
  private NOTAS: number[];
  private MEDIA: number;

  public NOME: string;

  constructor() {
    this.NOME = "";
    this.NOTAS = [];
    this.MEDIA = 0;
  }

  public CMEDIA(): number {
    let SOMA = 0;
    for (let I = 0; I < 4; I++) {
      SOMA += this.NOTAS[I];
    }
    this.MEDIA = SOMA / 4;
    return this.MEDIA;
  }

  public PEGANOTA(POS: number): number {
    return this.NOTAS[POS];
  }

  public PEGAMEDIA(): number {
    return this.CMEDIA();
  }

  public POENOTA(NT: number, POS: number): void {
    this.NOTAS[POS] = NT;
  }
}

const ALUNO: CLS_ALUNO = new CLS_ALUNO();

let I: number;
let ENTRANOTA: number;

console.log("DADOS ESCOLARES");
console.log();

const readline = require('readline-sync');

ALUNO.NOME = readline.question("Informe o nome: ");

console.log("Informe as notas:");
for (I = 0; I <= 3; I++) {
  ENTRANOTA = parseFloat(readline.question(`${I + 1}a. nota: `));
  ALUNO.POENOTA(ENTRANOTA, I);
}

console.log();

console.log("RELATORIO ESCOLAR");
console.log();

console.log("Nome: " + ALUNO.NOME);

for (I = 0; I <= 3; I++) {
  console.log(`${I + 1}a. nota: ${ALUNO.PEGANOTA(I)}`);
}
console.log("Media: " + ALUNO.PEGAMEDIA().toFixed(1));
